src/btree/bt_read.c

```diff
@@ -400,9 +400,15 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
                     evict_skip = true;
                 else if (ret == EBUSY) {
                     WT_NOT_READ(ret, 0);
-                    WT_STAT_CONN_INCR(session, page_forcible_evict_blocked);
-                    stalled = true;
-                    break;
+                    /* Do not retry to evict the pages during reconciliation. */
+                    if (F_ISSET(session, WT_SESSION_NO_RECONCILE)) {
```

Comment:
Could you update the comment here, the flag is generic and says "no reconcile", so the comment is confusing as it mentions not retrying during reconciliation.

Something like: "Don't back off if the session is configured not to do reconciliation, that just wastes time for no benefit. Without this check a reconciliation of a page that requires writing content to the history store can stall trying to force-evict a history store page when there is no chance it will be evicted"

Reply:
Thanks @agorrod. Fix in 25dee27a5e48da149414c2728d7eb2c281ac6c33